verifyFiles.py

```diff
@@ -198,25 +199,35 @@ def run_command(command,file_name,condition1,condition2,subject_name,root_dir):
     return output
 
 
-#Method is_checksum_equal reads the checksum
-#and returns true if the checksums are matching.
-def is_checksum_equal(condition_one_checksums_after_file,condition_two_checksums_after_file,subject,file_name):
-    file_name = "exec/"+subject+"/"+file_name
-    checksum_first_file=read_checksum_from_file(condition_one_checksums_after_file,file_name)
-    checksum_second_file=read_checksum_from_file(condition_two_checksums_after_file,file_name)
-    if checksum_first_file == None or checksum_second_file == None:
-        return True
-    return checksum_second_file==checksum_first_file
 
 #Method read_checksum_from_file gets the file path containing the checksum and the file name.
-#It reads the content line by line and if a match is found, it returns the checksum value.
-def read_checksum_from_file(checksums_after_file_path,file_name):
-    checksum=None
+#It reads the content line by line and if a match is found, it adds the file name as key and checksum as the value and returns dictionary after 
+#the file gets read.
+def read_checksum_from_file(checksums_after_file_path):
+    checksum_from_file_dict={}
     with open(checksums_after_file_path) as file:
          for line in file:
-	     if file_name in line:
-	         checksum = line.split(' ', 1)[0]
-    return checksum
+	     if "exec" in line:
+	         checksum_from_file_dict[(line.split(' ', 1)[1]).strip()]=(line.split(' ', 1)[0]).strip()
```

Comment:
You should check that the line has indeed 2 elements before trying to access them or a badly-formatted file will crash your code.